data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/OpenSearchService.java

```diff
@@ -4,14 +4,55 @@
  */
 package org.opensearch.dataprepper.plugins.source.opensearch;
 
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.source.coordinator.SourceCoordinator;
+import org.opensearch.dataprepper.plugins.source.opensearch.worker.PitWorker;
+import org.opensearch.dataprepper.plugins.source.opensearch.worker.ScrollWorker;
+import org.opensearch.dataprepper.plugins.source.opensearch.worker.client.SearchAccessor;
+
 public class OpenSearchService {
 
+    private final SearchAccessor searchAccessor;
+    private final OpenSearchSourceConfiguration openSearchSourceConfiguration;
+    private final SourceCoordinator<OpenSearchIndexProgressState> sourceCoordinator;
+    private final Buffer<Record<Event>> buffer;
+
+    private Thread searchWorkerThread;
+
+    public static OpenSearchService createOpenSearchService(final SearchAccessor searchAccessor,
+                                                            final SourceCoordinator<OpenSearchIndexProgressState> sourceCoordinator,
+                                                            final OpenSearchSourceConfiguration openSearchSourceConfiguration,
+                                                            final Buffer<Record<Event>> buffer) {
+        return new OpenSearchService(searchAccessor, sourceCoordinator, openSearchSourceConfiguration, buffer);
+    }
+
+    private OpenSearchService(final SearchAccessor searchAccessor,
+                             final SourceCoordinator<OpenSearchIndexProgressState> sourceCoordinator,
+                             final OpenSearchSourceConfiguration openSearchSourceConfiguration,
+                             final Buffer<Record<Event>> buffer) {
+        this.searchAccessor = searchAccessor;
+        this.openSearchSourceConfiguration = openSearchSourceConfiguration;
+        this.buffer = buffer;
+        this.sourceCoordinator = sourceCoordinator;
+        this.sourceCoordinator.initialize();
+    }
+
     public void start() {
-        // todo: to implement
-        // Leverages a runnable (SearchWorker) to perform the querying on the source cluster
+        switch(searchAccessor.getSearchContextType()) {
+            case POINT_IN_TIME:
+                searchWorkerThread = new Thread(new PitWorker(searchAccessor, openSearchSourceConfiguration, sourceCoordinator, buffer));
+                break;
+            case SCROLL:
+                searchWorkerThread = new Thread(new ScrollWorker(searchAccessor, openSearchSourceConfiguration, sourceCoordinator, buffer));
+                break;
+        }
```

Comment:
For a future CR, let's pull this logic into a strategy class that is deleted in this PR and do something like this here. It will be easier to test this control logic separate from thread creation:

```
final SearchWorker worker = SearchWorkerStrategy.get(searchAccessor, ...);
searchWorkerThread = new Thread(worker);
searchWorkerThread.start()
```